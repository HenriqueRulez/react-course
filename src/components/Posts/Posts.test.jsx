import { render, screen } from "@testing-library/react";
import { Posts } from ".";

const props = {
  posts: [
    {
      id: 1,
      title: "Title 1",
      body: "Body 1",
      cover: "img/img.png"
    },
    {
      id: 2,
      title: "Title 2",
      body: "Body 2",
      cover: "img/img2.png"
    },
    {
      id: 3,
      title: "Title 3",
      body: "Body 3",
      cover: "img/img3.png"
    },
  ]
}

describe('<Posts />', () => {
  it('should render posts', () => {
    render(<Posts {...props}/>);

    expect(screen.getAllByRole('heading', { name: /title/i })).toHaveLength(3);
    expect(screen.getAllByRole('heading', { name: /title/i })).toHaveLength(3);
    expect(screen.getAllByText(/body/i)).toHaveLength(3);
    expect(screen.getByRole('img', { name: /title 3/i })).toHaveAttribute('src', 'img/img3.png');
  });

  it('should match snapshot', () => {
    const { container } = render(<Posts {...props}/>);

    // eslint-disable-next-line testing-library/no-node-access
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should not render posts', () => {
    render(<Posts />);

    expect(screen.queryAllByRole('heading', { name: /title/i })).toHaveLength(0);

  });
})