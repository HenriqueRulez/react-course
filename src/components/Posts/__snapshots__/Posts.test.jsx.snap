// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts /> should match snapshot 1`] = `
<article
  class="posts"
>
  <div
    class="post"
  >
    <img
      alt="Title 1"
      src="img/img.png"
    />
    <section
      class="post-content"
    >
      <h2>
        Title 1
      </h2>
      <p>
        Body 1
      </p>
    </section>
  </div>
  <div
    class="post"
  >
    <img
      alt="Title 2"
      src="img/img2.png"
    />
    <section
      class="post-content"
    >
      <h2>
        Title 2
      </h2>
      <p>
        Body 2
      </p>
    </section>
  </div>
  <div
    class="post"
  >
    <img
      alt="Title 3"
      src="img/img3.png"
    />
    <section
      class="post-content"
    >
      <h2>
        Title 3
      </h2>
      <p>
        Body 3
      </p>
    </section>
  </div>
</article>
`;
